### PR TITLE
window state manager

### PR DESCRIPTION
**Pull request by stweyel**
#512
Merged electron-window-state-manager package.
Providing store/restore of window position.

### CODE REVIEW

**Comment by angle943** [comment]
@stweyel i've adjusted the window, then closed the app by clicking "x", then reopened it agian, but it doesn't seem to save the window state?

**Comment by stweyel** [comment]
> @stweyel i've adjusted the window, then closed the app by clicking "x", then reopened it agian, but it doesn't seem to save the window state?

hmm... that's strange. Working fine here on MacOS and Windows10. What are your system specs?

**Comment by angle943** [comment]
@stweyel what is the filepath of where the `json` file should be located on a mac?

**Comment by stweyel** [comment]
> @stweyel what is the filepath of where the `json` file should be located on a mac?

@angle943 at first look it should be in Library/Application Support/. But it's not there. I'll check where else it could be saved.
However, if you tested on mac how did you get your window to reopen after closing it with 'x' without quitting the app?
Still have no clue why it fails on your side. Have it running on three different MacOS

**Comment by stweyel** [comment]
@angle943 found it! It's in the users library folder:
/Users/.../Library/Application Support/TNB Account Manager/windowStates.json

**Comment by angle943** [comment]
nope, i don't see windowStates.json there :/ I have MacOS Big Sur

**Comment by stweyel** [comment]
Then for now i've no idea what's going on. It's perfectly running here on BigSur and i can't reproduce the fail. Well in fact there  isn't very much what could fail. Probably also  no write permission issue because without that the account manager doesn't even start. 
Maybe  you can check if the dependencies were properly installed  (electron-window-state-manager, app-root-path)?
Another thing to try is to manually put the attached file (dezip first!) into the TNB Account Manager Application Support folder and see if it gets loaded...
However i'll investigate further.., Maybe @tomijaga could give it a try on his mac?

[windowStates.json.zip](https://github.com/thenewboston-developers/Account-Manager/files/6292235/windowStates.json.zip)

**Comment by angle943** [comment]
@thesanjeevsharma @jamessspanggg @Rajputusman  could you give this a try?

**Comment by stweyel** [comment]
@angle943 since nobody replied...
Maybe just a last try. If you don't mind you could run the latest app build from me. I uploaded the alpha.45 dmg including my PR. If that would work on your side maybe something in the PR is corrupt (i just noticed the 'electron-window-state-manager.d.ts' is empty) and i'd open a new one.
Here the link to my build: https://www.dropbox.com/s/1zyjk2vhwn4egcd/TNB%20Account%20Manager-1.0.0-alpha.45-mac.dmg?dl=0

**Comment by jamessspanggg** [comment]
> @angle943 since nobody replied...
> Maybe just a last try. If you don't mind you could run the latest app build from me. I uploaded the alpha.45 dmg including my PR. If that would work on your side maybe something in the PR is corrupt (i just noticed the 'electron-window-state-manager.d.ts' is empty) and i'd open a new one.
> Here the link to my build: https://www.dropbox.com/s/1zyjk2vhwn4egcd/TNB%20Account%20Manager-1.0.0-alpha.45-mac.dmg?dl=0

Sorry for the later response! I missed it. Yes the app you provided is working fine with the window size and position restored 👍🏻 using macOS Big Sur 11.2.3

File: package.json [hunk 1]
```diff
@@ -123,6 +123,7 @@
   "dependencies": {
     "@thenewboston/ui": "^0.1.5",
     "@thenewboston/utils": "^1.0.0",
+    "app-root-path": "3.0.0",
```
Comment (jamessspanggg) [inline package.json:126]: is `app-root-path` required? it seems like it is under dependency of `electron-window-state-manager`

Screen Shot 2021-04-18 at 9 10 55 AM

Comment (stweyel) [reply package.json:126]: Glad it's working!. Yes app-root-path is needed by window-state-manager. Though it's installed as an own dependency i got errors and therefore added it to the main tree.

**Comment by stweyel** [comment]
> Sorry for the later response! I missed it. Yes the app you provided is working fine with the window size and position restored 👍🏻 using macOS Big Sur 11.2.3

No prob. Glad it's working! Shall i open a new PR?

**Comment by jamessspanggg** [comment]
> No prob. Glad it's working! Shall i open a new PR?

Why so? I think we can keep the PR

**Review by jamessspanggg — APPROVED** [review]

**Review by sno2 — APPROVED** [review]

**Comment by angle943** [comment]
ok we'll merge it in

**Pull request merged**